Web Quiz Engine (Kotlin)/task/src/engine/WebQuizApiController.kt

```diff
@@ -1,69 +1,46 @@
 package engine
 
+import engine.entity.*
+import engine.service.QuizService
 import jakarta.validation.Valid
-import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.http.HttpStatus
 import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.*
 
-
-// exception dans leur propre package
-// pas de logique dans POJO
-// chaque POJO dans son ficher dans un package model ?
-// inclue repsonse et request body
-// pas de logique dans le controller
-
-
-data class AnswerQuizRequestBody(val answer: List<Int>?)
 @RestController
-class WebQuizApiController {
-
-    @Autowired
-    private lateinit var quizRepository: WebQuizRepository
-
-    sealed class QuizPostResponseBody(val success: Boolean, val feedback: String)
-    class QuizPostResponseBodyCorrect(success: Boolean = true, feedback: String = "Congratulations, you're right!")
-        : QuizPostResponseBody(success, feedback)
-    class QuizPostResponseBodyIncorrect(success: Boolean = false, feedback: String = "Wrong answer! Please, try again.")
-        : QuizPostResponseBody(success, feedback)
+class WebQuizApiController(private val quizService: QuizService) {
 
     @PostMapping("/api/quizzes/{quizId}/solve")
```

Comment:
You can add `@ResposeStatus(HttpStatus.OK)`.
You don't have any logic to return a different HTTP code so it's better to define it here instead of making the Controller return a response entity with a code.

Reply:
Pretty cool. Done. Here and for every endpoint